src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -1006,78 +1009,70 @@ ForeignFilePath(const char *nodeName, uint32 nodePort, const char *tableName)
 /*
  * ExecuteRemoteQuery executes the given query, copies the query's results to a
  * sorted list, and returns this list. The function assumes that query results
- * have a single column, and asserts on that assumption. If results are empty,
- * or an error occurs during query runtime, the function returns an empty list.
- * If asUser is NULL the connection is established as the current user,
+ * have a single column and ignores other columns there are more than one. If results
+ * are empty or an error occurs during query runtime, the function returns an empty
+ * list. If asUser is NULL, the connection is established as the current user,
  * otherwise as the specified user.
  */
 List *
 ExecuteRemoteQuery(const char *nodeName, uint32 nodePort, char *runAsUser,
 				   StringInfo queryString)
 {
-	int32 connectionId = -1;
-	bool querySent = false;
-	bool queryReady = false;
-	bool queryOK = false;
-	void *queryResult = NULL;
+	MultiConnection *connection = NULL;
+	uint32 connectionFlags = 0;
+	PGresult *queryResult = NULL;
+	ExecStatusType resultStatus = PGRES_COMMAND_OK;
 	int rowCount = 0;
 	int rowIndex = 0;
-	int columnCount = 0;
+	int querySent = 0;
 	List *resultList = NIL;
 
-	connectionId = MultiClientConnect(nodeName, nodePort, NULL, runAsUser);
-	if (connectionId == INVALID_CONNECTION_ID)
+	/* get the connection to the node */
+	connection = GetNodeUserDatabaseConnection(connectionFlags, nodeName, nodePort,
+											   runAsUser, NULL);
+
+	/* check the status of the connection */
```

Comment:
Added that but it affected the regression tests in a strange way. I am not sure if these are symptoms of a more serious change.